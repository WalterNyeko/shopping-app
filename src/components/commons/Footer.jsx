import React from "react";
import { NavLink } from "react-router-dom";

const Footer = ({ departments }) => {
  return (
    <footer className="footer bg-dark sticky-bottom mb-auto py-3">
      <div className="container text-center h5 mt-4">
        {departments &&
          departments.length &&
          departments.map(({ name, department_id }) => (
            <NavLink
              to={`/department/${department_id}`}
              key={department_id}
              className="text-white mr-5"
            >
              {name}
            </NavLink>
          ))}
      </div>

      <div className="container text-center mt-4 mb-4">
        <a className="text-white ml-5 mr-5 mb-4" href="#">
          <i className="fab fa-instagram fa-3x" />
        </a>
        <a className="text-white ml-5 mr-5 mb-4" href="#">
          <i className="fab fa-pinterest fa-3x" />
        </a>
        <a className="text-white ml-5 mr-5 mb-4" href="#">
          <i className="fab fa-twitter fa-3x" />
        </a>
        <a className="text-white ml-5 mr-5 mb-4" href="#">
          <i className="fab fa-facebook fa-3x" />
        </a>
      </div>

      <div className="container text-center mt-4 mb-4">
        <a className="text-white ml-5 mr-5 mb-4" href="#">
          &copy; 2019 Shopmate Ltd
        </a>
        <a className="text-white ml-5 mr-5 mb-4" href="#">
          Contact
        </a>
        <a className="text-white ml-5 mr-5 mb-4" href="#">
          Privacy policy
        </a>
      </div>
    </footer>
  );
};
export default Footer;
