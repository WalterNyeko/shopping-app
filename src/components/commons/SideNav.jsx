import React, { Fragment } from 'react';
import '../../styles/SideNav.css';

 const SideNav = (props) => {
     const { content } = props;
    return (
        <Fragment>
            <div className="row">
                <div className="col-md-2">
                    <div className="sidenav">
                        <h6>Categories</h6>
                        <hr></hr>
                        <a href="#">Christmas</a>
                        <a href="#">Valentines</a>
                        <a href="#">Ladies</a>
                        <a href="#">Children</a>
                    </div>
                </div>
                <div className="col-md-10">
                    {content}
                </div>

            </div>
        </Fragment>
    )
}
export default SideNav;
