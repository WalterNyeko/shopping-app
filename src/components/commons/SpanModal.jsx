import React, { Fragment } from 'react';
import { Modal} from 'antd';



const ReusableModal = (props) => {
    const { 
        spanText, 
        modalTitle, 
        modalContent, 
        modalWidth,
        handleOk, 
        handleCancel,
        showModal,
        visible,
        confirmLoading,
        className,
    } = props;
    return (
      <Fragment>
        <span onClick={showModal} className={className}>
            {spanText}
        </span>
        <Modal
            title={modalTitle}
            visible={visible}
            onOk={handleOk}
            confirmLoading={confirmLoading}
            onCancel={handleCancel}
            width={modalWidth}
            footer={null}
        >
            <p>{modalContent}</p>
        </Modal>
      </Fragment>
    );
}
export default ReusableModal;