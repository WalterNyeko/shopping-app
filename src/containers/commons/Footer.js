import React, { Component, Fragment } from "react";
import FooterPage from "../../components/commons/Footer";

class Footer extends Component {
  render() {
    const { departments } = this.props;
    return (
      <Fragment>
        <FooterPage departments={departments} />
      </Fragment>
    );
  }
}
export default Footer;
