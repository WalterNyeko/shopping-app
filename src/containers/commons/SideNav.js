import React, { Component, Fragment } from 'react';
import SideNavComponent from '../../components/commons/SideNav';

class SideNav extends Component {
    render() {
        const { content } = this.props;
        return (
            <Fragment>
                <SideNavComponent content={content}/>
            </Fragment>
        )
    }
}
export default SideNav;
