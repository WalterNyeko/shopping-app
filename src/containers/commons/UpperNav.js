import React, { Component, Fragment } from "react";
import { connect } from "react-redux";
import UpperNavComponent from "../../components/commons/UpperNav";
import { getDepartments } from "../../store/actions/Departments";
import {
  getTotalAmount,
  getItemsInCart,
  emptyShoppingCart
} from "../../store/actions/Orders";
import history from "../../helpers/history";

export class UpperNav extends Component {
  state = {
    isLoggedIn: false
  };

  /**
   * 1. retrieves departments, total amount and items in shopping cart
   * 2. sets the state value of isLoggedIn to false
   *
   * @param {Object} props
   *
   * @returns {void}
   */
  retrieveItems = props => {
    const jwtToken = localStorage.getItem("jwt-token");
    const { getDepartments, getTotalAmount, getItemsInCart } = props;
    jwtToken && this.setState({ isLoggedIn: true });
    getDepartments();
    const cartId = localStorage.getItem("cartId");
    cartId && getTotalAmount(cartId);
    getItemsInCart(cartId);
  };

  componentWillMount() {
    return this.retrieveItems(this.props);
  }

  componentWillReceiveProps(nextProps) {
    if (
      nextProps.myCart.totalAmountOfItemsInCart.total_amount !==
      this.props.myCart.totalAmountOfItemsInCart.total_amount
    ) {
      return this.retrieveItems(nextProps);
    }
  }

  /**
   * logs out a user from the app by doing the following
   * 1. clears the users token from localStorage
   * 2. clears the users name from loaclStorage
   * 3. redirects user to the landing page
   *
   * @returns {void}
   */
  handleLogout = () => {
    localStorage.removeItem("jwt-token");
    localStorage.removeItem("user");
    const cartId = localStorage.getItem("cartId");
    const {
      emptyShoppingCart,
      myCart: { cartItems }
    } = this.props;
    cartItems && cartItems.length && emptyShoppingCart(cartId);
    const { pathname } = history.location;
    if (pathname === "/") {
      history.push("/home");
    } else {
      history.push("/");
    }
  };
  render() {
    const { isLoggedIn } = this.state;
    const {
      allDepartments,
      myCart: {
        totalAmountOfItemsInCart: { total_amount },
        cartItems
      }
    } = this.props;
    return (
      <Fragment>
        <UpperNavComponent
          isLoggedIn={isLoggedIn}
          allDepartments={allDepartments}
          totalAmount={total_amount}
          handleLogout={this.handleLogout}
          cartItems={cartItems}
        />
      </Fragment>
    );
  }
}
const mapStateToProps = state => ({
  allDepartments: state.departments,
  myCart: state.itemsInCart
});
export default connect(
  mapStateToProps,
  { getDepartments, getTotalAmount, emptyShoppingCart, getItemsInCart }
)(UpperNav);
