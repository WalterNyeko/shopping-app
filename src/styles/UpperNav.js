import { makeStyles } from "@material-ui/core/styles";
import { fade } from "@material-ui/core/styles/colorManipulator";

export const useStyles = makeStyles(theme => ({
  root: {
    flexGrow: 1,
    overflow: "hidden",
    top: 0 /* Position the navbar at the top of the page */,
    width: "100%" /* Full width */,
    position: "fixed",
    zIndex: 1
  },
  bar: {
    boxShadow: "none",
    height: "40px",
    paddingLeft: "20px",
    paddingBottom: "5px",
    paddingTop: "5px"
  },
  bigBar: {
    boxShadow: "none",
    height: "80px",
    backgroundColor: "#343434",
    color: "white",
    padding: "10px"
  },

  navItems: {
    color: "red",
    fontWeight: "14px",
    textDecoration: "none",
    cursor: "pointer",
    // zIndex: "5035"
  },

  navItemsMainBar: {
    color: "white",
    fontWeight: "18px",
    textDecoration: "none",
    cursor: "pointer"
  },

  defaultNavItems: {
    color: "gray",
    fontWeight: "14px",
    textDecoration: "none"
  },

  rightMostItems: {
    color: "gray",
    float: "right",
    fontWeight: "14px",
    textDecoration: "none",
    marginRight: "10px"
  },
  centralItems: {
    color: "gray",
    alignContent: "center",
    fontWeight: "15px",
    textDecoration: "none"
  },
  bigBarToolBar: {
    width: "100vw"
  },

  shopmateLogo: {
    color: "#ff3d3d",
    alignContent: "center",
    fontSize: "35px",
    textDecoration: "none"
  },
  menuButton: {
    marginRight: theme.spacing(2)
  },
  title: {
    display: "none",
    [theme.breakpoints.up("sm")]: {
      display: "block"
    }
  },
  search: {
    position: "relative",
    borderRadius: theme.shape.borderRadius,
    backgroundColor: fade(theme.palette.common.white, 0.15),
    "&:hover": {
      backgroundColor: fade(theme.palette.common.white, 0.25)
    },
    marginRight: theme.spacing(2),
    marginLeft: 0,
    width: "100%",
    [theme.breakpoints.up("sm")]: {
      marginLeft: theme.spacing(3),
      width: "auto"
    }
  },
  searchIcon: {
    width: theme.spacing(7),
    height: "100%",
    position: "absolute",
    pointerEvents: "none",
    display: "flex",
    alignItems: "center",
    justifyContent: "center"
  },
  inputRoot: {
    color: "inherit"
  },
  inputInput: {
    padding: theme.spacing(1, 1, 1, 7),
    transition: theme.transitions.create("width"),
    width: "100%",
    [theme.breakpoints.up("md")]: {
      width: 200
    }
  },
  sectionDesktop: {
    display: "none",
    [theme.breakpoints.up("md")]: {
      display: "flex"
    }
  },
  sectionMobile: {
    display: "flex",
    [theme.breakpoints.up("md")]: {
      display: "none"
    }
  },

  searchBoxPadding: {
    padding: "10px"
  }
}));
