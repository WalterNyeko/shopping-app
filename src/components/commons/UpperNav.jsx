import React, { Fragment } from "react";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import { NavLink } from "react-router-dom";
import bag from "../../images/shopping-bag.png";
import SearchIcon from "@material-ui/icons/Search";
import InputBase from "@material-ui/core/InputBase";
import { useStyles } from "../../styles/UpperNav";
import { SpanModal } from "../../../src/containers/commons/SpanModal";
import SignUpPage from "../../containers/customers/SignupPage";
import LoginPage from "../../containers/customers/LoginPage";

const UpperNav = ({ isLoggedIn, allDepartments, totalAmount }) => {
  const classes = useStyles();
  const { departments } = allDepartments;
  return (
    <Fragment>
      <nav className="navbar navbar-expand-md sticky-top navbar-light bg-light pt-0 pb-0 fir">
        <div className="navbar-brand">
          Hey!{" "}
          <a className="text-danger" href="#">
            <SpanModal
              spanText="Sign In"
              modalTitle="Login To Your ShopMate Account"
              modalWidth="450px"
              modalContent={<LoginPage />}
              className={classes.navItems}
            />
          </a>{" "}
          or{" "}
          <a href="#" className="text-danger">
            <SpanModal
              spanText="Register"
              modalTitle="Register For A ShopMate"
              modalWidth="450px"
              modalContent={<SignUpPage />}
              className={classes.navItems}
            />
          </a>
        </div>
        <button
          className="navbar-toggler"
          type="button"
          data-toggle="collapse"
          data-target="#navbarsExample04"
          aria-controls="navbarsExample05"
          aria-expanded="false"
          aria-label="Toggle navigation"
        >
          <span className="navbar-toggler-icon" />
        </button>
        <div className="collapse navbar-collapse" id="navbarsExample04">
          <ul className="navbar-nav m-auto nav-list">
            <li className="nav-item active">
              <a className="nav-link" href="#">
                Daily Deals <span className="sr-only">(current)</span>
              </a>
            </li>
            <li className="nav-item active">
              <a className="nav-link" href="#">
                Sell
              </a>
            </li>
            <li className="nav-item active">
              <a className="nav-link" href="#">
                Help & Contact
              </a>
            </li>
          </ul>
          <form className="form-inline my-2 my-md-0">
            <NavLink to="/mycart/" className="my-bag">
              <img src={bag} alt="bag" /> &nbsp; Your Bag &nbsp;{" "}
              <span className="text-danger"> $ {totalAmount}</span>
            </NavLink>
          </form>
        </div>
      </nav>
      <nav className="navbar navbar-expand-md fixed-top navbar-dark bg-dark nav-margin">
        <a className="navbar-brand logo" href="#">
          <NavLink to="/" className={classes.shopmateLogo}>
            SHOPMATE
          </NavLink>
        </a>
        <button
          className="navbar-toggler"
          type="button"
          data-toggle="collapse"
          data-target="#navbarsExample05"
          aria-controls="navbarsExample05"
          aria-expanded="false"
          aria-label="Toggle navigation"
        >
          <span className="navbar-toggler-icon" />
        </button>
        <div className="collapse navbar-collapse" id="navbarsExample05">
          <ul className="navbar-nav m-auto">
            {departments &&
              departments.map(({ department_id, name }) => (
                <Fragment key={department_id}>
                  <NavLink
                    to={`/department/${department_id}`}
                    className={classes.navItemsMainBar}
                  >
                    <li className="nav-item active">
                      <a className="nav-link nav-link-item" href="#">
                        {name}
                        <span className="sr-only">(current)</span>
                      </a>
                    </li>
                  </NavLink>
                  &nbsp;&nbsp;&nbsp;&nbsp;
                </Fragment>
              ))}
          </ul>
          <form className="form-inline my-2 my-md-0 text-white">
            <div className={classes.search}>
              <div className={classes.searchIcon}>
                <SearchIcon />
              </div>
              <InputBase
                placeholder="Searchâ€¦"
                classes={{
                  root: classes.inputRoot,
                  input: classes.inputInput
                }}
              />
            </div>
          </form>
        </div>
      </nav>
    </Fragment>
  );
};
export default UpperNav;
