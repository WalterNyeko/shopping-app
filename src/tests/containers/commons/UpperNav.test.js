import React from "react";
import { UpperNav } from "../../../containers/commons/UpperNav";
import { mount } from "enzyme";
import { Provider } from "react-redux";
import { Router } from "react-router";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import history from "../../../helpers/history";

const initialState = {
  allDepartments: { departments: [{ department_id: 1, name: "Test" }] },
  myCart: {
    totalAmountOfItemsInCart: { total_amount: 10.0 },
    cartItems: [{}]
  }
};
const props = {
  getDepartments: jest.fn(),
  getItemsInCart: jest.fn(),
  myCart: {
    totalAmountOfItemsInCart: {
      total_amount: 10
    }
  },
  allDepartments: {
    departments: [
      {
        department_id: 1,
        name: "Test"
      }
    ]
  }
};
const mockStore = configureStore([thunk]);
const store = mockStore(initialState);
describe("UpperNav", () => {
  const wrapper = mount(
    <Router history={history}>
      <UpperNav {...props} />
    </Router>
  );
  it("should render side navigation without crashing", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
