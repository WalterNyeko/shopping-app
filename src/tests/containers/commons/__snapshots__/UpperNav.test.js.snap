// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpperNav should render side navigation without crashing 1`] = `ReactWrapper {}`;
